### ISSUE TITLE
Tutorial 4: question about a comment in forward() in Encoder()

### ISSUE THREAD

**Issue by byc1234** [bentrevett/pytorch-seq2seq#137]
`forward()` in `Encoder` has a comment(`#src_len = [batch size]`).
Shouldn't `[batch size]` be `[src len]` instead?
I see your explanation above `def train()` and src_len within the mothod is **the lengths of each sentence within the batch**.
```
As we are using include_lengths = True for our source field, batch.src is now a tuple with the first element being the numericalized tensor representing the sentence and the second element being the lengths of each sentence within the batch.
```

Also https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pack_padded_sequence.html shows
```
torch.nn.utils.rnn.pack_padded_sequence(input, lengths, batch_first=False, enforce_sorted=True)

input can be of size T x B x * where T is the length of the longest sequence (equal to lengths[0]), B is the batch size, and * is any number of dimensions (including 0). If batch_first is True, B x T x * input is expected.
```

**Comment by bentrevett** [comment]
The `src_len` tensor should one dimension with `batch_size` elements, each element is the length (an integer) of how long that sequence is without padding.

Consider if we had a batch size of 4 and sequence lengths of 6, 8, 10, 12. Thus, we'd have a four-element tensor, **[6, 8, 10, 12]**. The shape of this is (4), which is equal to the batch size, therefore the tensor is size **[batch_size]**.

Let me know if that clears things up.

**Comment by byc1234** [comment]
Thanks for the detailed explanation.
